tokio/src/signal/windows.rs

```diff
@@ -175,43 +175,43 @@ impl Stream for CtrlBreak {
 #[cfg(test)]
 mod tests {
     use super::*;
+    use crate::runtime::current_thread::Runtime;
 
-    use tokio::timer::Timeout;
-
-    use futures_util::future::FutureExt;
     use futures_util::stream::StreamExt;
-    use std::future::Future;
-    use std::time::Duration;
 
-    fn with_timeout<F: Future>(future: F) -> impl Future<Output = F::Output> {
-        Timeout::new(future, Duration::from_secs(1)).map(|result| result.expect("timed out"))
-    }
+    #[test]
+    fn ctrl_c() {
+        let mut rt = Runtime::new().unwrap();
 
-    #[tokio::test]
-    async fn ctrl_c() {
-        let ctrl_c = crate::signal::ctrl_c().expect("failed to create CtrlC");
+        rt.block_on(async {
+            let ctrl_c = crate::signal::ctrl_c().expect("failed to create CtrlC");
 
-        // Windows doesn't have a good programmatic way of sending events
-        // like sending signals on Unix, so we'll stub out the actual OS
-        // integration and test that our handling works.
-        unsafe {
-            super::handler(CTRL_C_EVENT);
-        }
+            // Windows doesn't have a good programmatic way of sending events
+            // like sending signals on Unix, so we'll stub out the actual OS
+            // integration and test that our handling works.
+            unsafe {
+                super::handler(CTRL_C_EVENT);
+            }
 
-        let _ = with_timeout(ctrl_c.into_future()).await;
+            let _ = ctrl_c.into_future().await;
+        });
     }
 
-    #[tokio::test]
-    async fn ctrl_break() {
-        let ctrl_break = super::ctrl_break().expect("failed to create CtrlC");
+    #[test]
```

Comment:
Just curious, why the switch away from `#[tokio::test]`? Seems like there wasn't much of a change??

Reply:
`tokio::test` doesn't trivially work when called from the tokio crate. There probably is a way to fix the macro, but I didn't want to spent the time in the PR.

The issue is the runtime can't be referenced as `tokio::runtime::....`

Reply:
Ah, good to know, thanks